### DISCUSSION TITLE
接入钉钉飞书扫码登录方案

### DISCUSSION THREAD

**Discussion by gosszhang** [go-atomci/atomci#166] [Ideas]
#63 
实现方案：
1. 一个配置表，一个用户表
2. 登录页面根据配置开启情况，增加扫码登录按钮
3. 增加钉钉扫码登录配置页面和飞书扫码登录配置页面
4. 增加授权用户（钉钉/飞书用户）列表

表结构如下：

CREATE TABLE `oauth_config` (
  `id` bigint NOT NULL AUTO_INCREMENT,
  `platform` tinyint NOT NULL DEFAULT '0' COMMENT '平台：1钉钉,2飞书',
  `enable` tinyint NOT NULL DEFAULT '0' COMMENT '是否启用',
  `app_key` varchar(255) NOT NULL DEFAULT '' COMMENT 'AppKey',
  `app_secret` varchar(255) NOT NULL DEFAULT '' COMMENT 'AppSecret',
  `auto_bind` tinyint NOT NULL DEFAULT '0' COMMENT '自动绑定系统账号,手机号与登录账号匹配',
  `auto_create_user` tinyint NOT NULL DEFAULT '0' COMMENT '自动创建账号',
  `auto_activate` tinyint NOT NULL DEFAULT '0' COMMENT '账号是否为激活状态,0需要手动激活，1自动激活',
  `create_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `update_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `delete_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `oauth_config_delete_at` (`delete_at`)
) ENGINE=InnoDB ;

CREATE TABLE `oauth_user` (
  `id` bigint NOT NULL AUTO_INCREMENT,
  `oauth_config_id` bigint NOT NULL DEFAULT 0 COMMENT '授权登录id',
  `sys_user_id` bigint NOT NULL DEFAULT 0 COMMENT '系统用户id',
  `name` varchar(255) NOT NULL DEFAULT '' COMMENT '姓名',
  `mobile` varchar(255) NOT NULL DEFAULT '' COMMENT '手机号',
  `union_id` varchar(255) NOT NULL DEFAULT '' COMMENT 'unionid',
  `active` tinyint NOT NULL DEFAULT '0' COMMENT '是否激活',
  `create_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `update_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `delete_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `oauth_config_union_id` (`oauth_config_id`,`union_id`),
  KEY `idx_sys_user_id` (`sys_user_id`),
  KEY `oauth_user_delete_at` (`delete_at`)
) ENGINE=InnoDB ;

**Comment by fanhousanbu** [comment]
1. 表1是否也可以考虑用配置文件的方案？
2. just do it

**Comment by colynn** [comment]
@gosszhang  支持对于oauth_config 通过表的形式来 管理配置；

有几个小问题，可以一起讨论下：
1. oauth_user  与 sys_user的关系 是怎样的, 一对一吗？
2. 与 `auth.go` interface的实现如何， https://github.com/go-atomci/atomci/blob/master/internal/core/auth/auth.go#L35
3. 期望的interface 是什么形式的， 比如未来接入企业微信等是否也是可以沿用这个inteface?
